src/bot/mtproto-api/workrooms/close-chat.ts

```diff
@@ -0,0 +1,139 @@
+import { Context, SupportedEvents } from "#root/types/context";
+import { CallbackResult } from "#root/types/proxy.js";
+import { MtProto } from "../bot/mtproto";
+import { Api } from "telegram";
+import bigInt from "big-integer";
+
+export async function closeChat(context: Context<"issues.closed", SupportedEvents["issues.closed"]>): Promise<CallbackResult> {
+  const {
+    payload,
+    adapters: {
+      supabase: { chats },
+    },
+    logger,
+  } = context;
+  const mtProto = new MtProto(context);
+  await mtProto.initialize();
+
+  logger.info("Closing chat with name: ", { chatName: payload.issue.title });
+  const chat = await chats.getChatByTaskNodeId(payload.issue.node_id);
+
+  const fetchChat = await mtProto.client.invoke(
+    new mtProto.api.messages.GetFullChat({
+      chatId: chat.chatId,
+    })
+  );
+
+  if (!fetchChat) {
+    throw new Error("Failed to fetch chat");
+  }
+
+  let chatParticipants;
+
+  if ("participants" in fetchChat.fullChat) {
+    chatParticipants = fetchChat.fullChat.participants;
+  } else {
+    throw new Error("Failed to fetch chat participants");
+  }
+
+  // archive it
+  await mtProto.client.invoke(
+    new mtProto.api.folders.EditPeerFolders({
+      folderPeers: [
+        new mtProto.api.InputFolderPeer({
+          peer: new mtProto.api.InputPeerChat({ chatId: chat.chatId }),
+          folderId: 1, // 0 is active, 1 is archived
```

Comment:
Pretty fancy.

Reply:
Cleaner than just deleting having a zoo of chats that's the account doesn't have access to but yeah pretty cool